### DISCUSSION TITLE
Начинаем разбираться с MSSQL

### DISCUSSION THREAD

**Discussion by pingvin1308** [cleannetcode/LessonMonitor#6] [MSSQL]
Для выполнения домашней работы, вам необходимо открыть и сделать форк основного репозитория по ссылке:
https://github.com/cleannetcode/LessonMonitor/tree/homework/sql%2F1

По выполнению работы, вам необходимо создать пулл реквест из вашей ветки с выполненной домашкой в основной репозиторий.

Почитать:
- 1, 2 и 3 главы метанит: https://metanit.com/sql/sqlserver/
- https://www.sqlservertutorial.net/getting-started/what-is-sql-server/
- https://www.sqlservertutorial.net/install-sql-server/
- https://www.sqlservertutorial.net/connect-to-the-sql-server/
- https://www.sqlservertutorial.net/sql-server-sample-database/

Сделать:

- Скачать и установить все необходимые инструменты и сервисы
- Создать проект для работы с БД.
- Смоделировать БД, для моделирования следует выбрать пять сущностей из списка:
  - Members
  - Lessons
  - Homeworks
  - RoadMap
  - MemberStatistics
  - Skills
  - Topics
  - Questions
  - Timecodes
  - Achievements
  - Announcements
- Создать БД с помощью visual studio

Проверить:
провести ревью кода любого чужого пулл реквеста относящегося к данной домашке.

**Comment by emedvedeva** [comment]
Роман, по ссылкам от Метанита и https://www.sqlservertutorial.net/install-sql-server/ указаны разные способы установки. В результате и компоненты устанавливаются разные. Каким способом устанавливать? 
От метанита (https://prnt.sc/11osigv  https://prnt.sc/11osljj)
или второй (https://prnt.sc/11osgwp  https://prnt.sc/11osmg8)

**Reply by pingvin1308** [reply]
Думаю для начала лучше использовать windows authentication mode

**Reply by pingvin1308** [reply]
В будущих занятиях я покажу как использовать второй вариант

**Reply by emedvedeva** [reply]
> В будущих занятиях я покажу как использовать второй вариант

Если я уже установила второй варинт, как мне переделать на первый?

**Reply by pingvin1308** [reply]
А получилось подключиться к sql server?
Там просто если брать второй вариант, то будет запрашиваться логин и пароль

**Reply by pingvin1308** [reply]
![image](https://user-images.githubusercontent.com/17434920/115212326-0f653280-a109-11eb-86e0-c5be88ac1368.png)
а так, вот тут выбрать Properties и в меню Security убедиться, что стоит Windows Authentication mode. В целом судя по тому, что я виду у себя в настройках, ничего делать и не нужно. Даже если сделать настройку входа по логину и паролю, windows authentication mode будет работь

![image](https://user-images.githubusercontent.com/17434920/115212398-22780280-a109-11eb-856c-f4a291be3703.png)

**Reply by emedvedeva** [reply]
Спасибо)

**Comment by kharchenkoMichael** [comment]
Как правильно называть таблицы? во множественном числе или в единственном?

**Comment by kharchenkoMichael** [comment]
Через дизайнер в вижуал студие не меняется тип данных, только через код
![image](https://user-images.githubusercontent.com/44605049/153227002-1655e602-007b-4a47-8e84-763a8fd32163.png)
выбираю NVARCHAR(50), но ничего не происходит

**Reply by kharchenkoMichael** [reply]
получилось изменить через дизайнер, только если вписывать туда тип, просто кликом не получилось
